torch/_dynamo/variables/builtin.py

```diff
@@ -233,7 +240,11 @@ def call_function(
 
         if (
             self.can_insert_in_graph()
-            and tensor_args
+            # Many operators that we allow into an FX graph work for both tensors and SymInts.
+            # For those that don't, we should be able to rely on SymInt erroring.
+            # (Alternative would be to directly read off the list of allowed SymInt operations,
+            # although I'm not sure what that buys us).
+            and (tensor_args or dyn_shape_args)
```

Comment:
My thought was that for builtins that we allow in the FX graph (e.g. builtins.neg), we'd ideally want to be able to reuse the same dynamo code for tensors and dynamic shapes as much as possible.

The only real difference (as I see it) is that SymInt supports a smaller list of builtins than Tensor - but I'm just relying on SymInt erroring properly. E.g. when I tried `x.shape[0] or x.shape[1]` with this PR, I get:
```
TypeError: unsupported operand type(s) for |: 'SymInt' and 'SymInt'
```

Reply:
The particular translation you quote here looks wrong. SymInt could in principle support bitwise or, and this would be completely the wrong thing to call if the user wrote logical or.